File: server/src/main/java/net/progruzovik/dissent/captain/Player.java
package net.progruzovik.dissent.captain;

import net.progruzovik.dissent.model.domain.CaptainStatus;
import net.progruzovik.dissent.model.message.Sender;
import org.springframework.lang.NonNull;

public interface Player extends Captain, Sender {

    @NonNull
    CaptainStatus getStatus();

    void addToQueue();

    void removeFromQueue();

    void startMission(int missionIndex);
}

File: server/src/main/kotlin/net/progruzovik/dissent/captain/Player.kt
package net.progruzovik.dissent.captain

import net.progruzovik.dissent.model.domain.CaptainStatus
import net.progruzovik.dissent.model.event.EventEmitter
import net.progruzovik.dissent.model.socket.ServerSubject

interface Player : Captain, EventEmitter<ServerSubject> {

    val status: CaptainStatus

    fun addToQueue()

    fun removeFromQueue()

    fun startMission(missionId: Int)
}
